Int8 packed weight serializer and deserializer

csummersea opened the pull request:
Summary: Add serializer/deserializer for int8 packed weights without architecture dependent packing

Test Plan:
```
buck test mode/opt caffe2/caffe2/quantization/server:fully_connected_dnnlowp_op_test
```

Differential Revision: D19689332

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 323d1772:


* **5/5** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=323d1772d74fa7a26ad0e2dbf7a70ecc7be5ca76).

### :detective: 4 new failures recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4413441) pytorch_linux_xenial_py3_clang5_asan_build (1/4)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=95171522) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=95171522))
<details>
<summary>
<code>Feb 04 20:31:29 caused by: Connection refused (os error 111)</code>
</summary>

```
Feb 04 20:31:29 +++ eval 'extract_trap_cmd ' 
Feb 04 20:31:29 ++++ extract_trap_cmd 
Feb 04 20:31:29 ++++ printf '%s\n' '' 
Feb 04 20:31:29 +++ printf '%s\n' cleanup 
Feb 04 20:31:29 ++ trap -- ' 
Feb 04 20:31:29 cleanup' EXIT 
Feb 04 20:31:29 ++ which sccache 
Feb 04 20:31:29 ++ sccache --stop-server 
Feb 04 20:31:29 Stopping sccache server... 
Feb 04 20:31:29 error: couldn't connect to server 
Feb 04 20:31:29 caused by: Connection refused (os error 111) 
Feb 04 20:31:29 ++ true 
Feb 04 20:31:29 ++ rm /var/lib/jenkins/sccache_error.log 
Feb 04 20:31:29 rm: cannot remove '/var/lib/jenkins/sccache_error.log': No such file or directory 
Feb 04 20:31:29 ++ true 
Feb 04 20:31:29 ++ SCCACHE_ERROR_LOG=/var/lib/jenkins/sccache_error.log 
Feb 04 20:31:29 ++ SCCACHE_IDLE_TIMEOUT=1200 
Feb 04 20:31:29 ++ RUST_LOG=sccache::server=error 
Feb 04 20:31:29 ++ sccache --start-server 
Feb 04 20:31:29 Starting sccache server... 
Feb 04 20:31:29 ++ sccache --zero-stats 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4413433) pytorch_linux_xenial_py2_7_9_build (2/4)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=95171776) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=95171776))
<details>
<summary>
<code>Feb 04 20:30:28 caused by: Connection refused (os error 111)</code>
</summary>

```
Feb 04 20:30:28 +++ eval 'extract_trap_cmd ' 
Feb 04 20:30:28 ++++ extract_trap_cmd 
Feb 04 20:30:28 ++++ printf '%s\n' '' 
Feb 04 20:30:28 +++ printf '%s\n' cleanup 
Feb 04 20:30:28 ++ trap -- ' 
Feb 04 20:30:28 cleanup' EXIT 
Feb 04 20:30:28 ++ which sccache 
Feb 04 20:30:28 ++ sccache --stop-server 
Feb 04 20:30:28 Stopping sccache server... 
Feb 04 20:30:28 error: couldn't connect to server 
Feb 04 20:30:28 caused by: Connection refused (os error 111) 
Feb 04 20:30:28 ++ true 
Feb 04 20:30:28 ++ rm /var/lib/jenkins/sccache_error.log 
Feb 04 20:30:28 rm: cannot remove '/var/lib/jenkins/sccache_error.log': No such file or directory 
Feb 04 20:30:28 ++ true 
Feb 04 20:30:28 ++ SCCACHE_ERROR_LOG=/var/lib/jenkins/sccache_error.log 
Feb 04 20:30:28 ++ SCCACHE_IDLE_TIMEOUT=1200 
Feb 04 20:30:28 ++ RUST_LOG=sccache::server=error 
Feb 04 20:30:28 ++ sccache --start-server 
Feb 04 20:30:28 Starting sccache server... 
Feb 04 20:30:29 ++ sccache --zero-stats 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4413442) pytorch_macos_10_13_py3_build (3/4)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=95171955) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=95171955))
<details>
<summary>
<code>Feb 04 12:35:50 FAILED: caffe2/CMakeFiles/torch_cpu.dir/quantization/server/fbgemm_pack_op.cc.o</code>
</summary>

```
Feb 04 12:35:50 ../c10/util/Registry.h:296:3: note: expanded from macro 'C10_REGISTER_CLASS' 
Feb 04 12:35:50   C10_REGISTER_TYPED_CLASS(RegistryName, #key, __VA_ARGS__) 
Feb 04 12:35:50   ^ 
Feb 04 12:35:50 ../c10/util/Registry.h:250:33: note: expanded from macro 'C10_REGISTER_TYPED_CLASS' 
Feb 04 12:35:50       Registerer##RegistryName::DefaultCreator<__VA_ARGS__>,                \ 
Feb 04 12:35:50                                 ^ 
Feb 04 12:35:50 3 warnings generated. 
Feb 04 12:35:50 [2333/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/quantization/server/fbgemm_pack_matrix_cache.cc.o 
Feb 04 12:35:50 [2334/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/quantization/server/elementwise_sum_relu_op.cc.o 
Feb 04 12:35:50 [2335/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/quantization/server/fbgemm_pack_op.cc.o 
Feb 04 12:35:50 FAILED: caffe2/CMakeFiles/torch_cpu.dir/quantization/server/fbgemm_pack_op.cc.o  
AVE_THREAD -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -fvisibility=hidden -O2 -DCAFFE2_BUILD_MAIN_LIB -DASMJIT_STATIC -std=gnu++14 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/quantization/server/fbgemm_pack_op.cc.o -MF caffe2/CMakeFiles/torch_cpu.dir/quantization/server/fbgemm_pack_op.cc.o.d -o caffe2/CMakeFiles/torch_cpu.dir/quantization/server/fbgemm_pack_op.cc.o -c ../caffe2/quantization/server/fbgemm_pack_op.cc 
Feb 04 12:35:50 ../caffe2/quantization/server/fbgemm_pack_op.cc:3:10: fatal error: 'caffe2/caffe2/quantization/server/proto/fbgemm_int8.pb.h' file not found 
Feb 04 12:35:50 #include "caffe2/caffe2/quantization/server/proto/fbgemm_int8.pb.h" 
Feb 04 12:35:50          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
Feb 04 12:35:50 1 error generated. 
Feb 04 12:35:51 [2336/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/quantization/server/fully_connected_dnnlowp_acc16_op.cc.o 
Feb 04 12:35:51 In file included from ../caffe2/quantization/server/fully_connected_dnnlowp_acc16_op.cc:1: 
Feb 04 12:35:51 In file included from ../caffe2/quantization/server/fully_connected_dnnlowp_acc16_op.h:3: 
Feb 04 12:35:51 In file included from ../caffe2/quantization/server/fully_connected_dnnlowp_op.h:5: 
Feb 04 12:35:51 In file included from ../caffe2/quantization/server/dnnlowp_op.h:9: 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4413444) pytorch_xla_linux_xenial_py3_6_clang7_build (4/4)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=95172255) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=95172255))
<details>
<summary>
<code>Feb 04 20:31:09 caused by: Connection refused (os error 111)</code>
</summary>

```
Feb 04 20:31:09 +++ eval 'extract_trap_cmd ' 
Feb 04 20:31:09 ++++ extract_trap_cmd 
Feb 04 20:31:09 ++++ printf '%s\n' '' 
Feb 04 20:31:09 +++ printf '%s\n' cleanup 
Feb 04 20:31:09 ++ trap -- ' 
Feb 04 20:31:09 cleanup' EXIT 
Feb 04 20:31:09 ++ which sccache 
Feb 04 20:31:09 ++ sccache --stop-server 
Feb 04 20:31:09 Stopping sccache server... 
Feb 04 20:31:09 error: couldn't connect to server 
Feb 04 20:31:09 caused by: Connection refused (os error 111) 
Feb 04 20:31:09 ++ true 
Feb 04 20:31:09 ++ rm /var/lib/jenkins/sccache_error.log 
Feb 04 20:31:09 rm: cannot remove '/var/lib/jenkins/sccache_error.log': No such file or directory 
Feb 04 20:31:09 ++ true 
Feb 04 20:31:09 ++ SCCACHE_ERROR_LOG=/var/lib/jenkins/sccache_error.log 
Feb 04 20:31:09 ++ SCCACHE_IDLE_TIMEOUT=1200 
Feb 04 20:31:09 ++ RUST_LOG=sccache::server=error 
Feb 04 20:31:09 ++ sccache --start-server 
Feb 04 20:31:09 Starting sccache server... 
Feb 04 20:31:09 ++ sccache --zero-stats 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

csummersea committed:
Int8 packed weight serializer and deserializer (#32925)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32925

Add serializer/deserializer for int8 packed weights without architecture dependent packing

Test Plan:
```
buck test mode/opt caffe2/caffe2/quantization/server:fully_connected_dnnlowp_op_test
```

Differential Revision: D19689332

fbshipit-source-id: c7354ca228cbbc5f14fa3bb25d867b0bb314d79e
